### PR TITLE
WDP-66

### PR DESCRIPTION
**Pull request by lukaszcwiklinski**
https://kindergarten.stxnext.pl/browse/WDP-66

### CODE REVIEW

File: src/presence_analyzer/tests.py [hunk 1]
```diff
@@ -38,20 +44,189 @@ def test_mainpage(self):
...
+            content_type='application/json',
+        )
+        self.assertEqual(
+            resp.status_code,
+            200,
+        )
+        data = json.loads(
+            resp.data,
+        )
+        # import ipdb; ipdb.set_trace()
```
Comment (piotrgrundas) [inline src/presence_analyzer/tests.py]: That's interesting.

File: src/presence_analyzer/tests.py [hunk 2]
```diff
@@ -38,20 +44,189 @@ def test_mainpage(self):
         """
         Test main page redirect.
         """
-        resp = self.client.get('/')
-        self.assertEqual(resp.status_code, 302)
+        resp = self.client.get(
+            '/',
+        )
```
Comment (piotrgrundas) [inline src/presence_analyzer/tests.py]: If it fits one line leave it.

File: src/presence_analyzer/tests.py [hunk 3]
```diff
@@ -2,13 +2,15 @@
 """
 Presence analyzer unit tests.
 """
-import os.path
+import os.path, sys
```
Comment (piotrgrundas) [inline src/presence_analyzer/tests.py]: https://www.python.org/dev/peps/pep-0008/#imports

File: src/presence_analyzer/tests.py [hunk 4]
```diff
@@ -86,6 +185,45 @@ def test_get_data(self):
...
+        """
+        self.assertEqual(utils.seconds_since_midnight(
+            datetime.time(10, 30, 50)), 37850)
+
+    def test_interval(self):
+        """
+        Test interval.
+        """
+        self.assertEqual(utils.interval(
+            datetime.time(10, 10, 10), datetime.time(10, 30, 50)), 1240)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Style

File: src/presence_analyzer/tests.py [hunk 5]
```diff
@@ -42,6 +44,97 @@ def test_mainpage(self):
...
+            u'Sun': 0,
+        }
+
+        resp = self.client.get(
+            '/api/v1/presence_weekday/10',
+            content_type='application/json',
+        )
+        self.assertEqual(resp.status_code, 200)
+        data = json.loads(resp.data)
+        # import ipdb; ipdb.set_trace()
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ????

File: src/presence_analyzer/tests.py [hunk 6]
```diff
@@ -42,6 +44,97 @@ def test_mainpage(self):
...
+        resp = self.client.get('/static/mean_time_weekday.html')
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn(b'Presence mean time by weekday', resp.data)
+
+    def mean_time_parameter_correct(self):
+        """
+        Test presence by weekday api route parameter is correct.
+        """
+        weekdays = {
+            u'Mon': 0,
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: from __future__ import unicode_literals

File: src/presence_analyzer/tests.py [hunk 7]
```diff
@@ -3,16 +3,18 @@
 Presence analyzer unit tests.
 """
 import os.path
+import sys
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Sort imports

File: src/presence_analyzer/tests.py [hunk 8]
```diff
@@ -3,16 +3,18 @@
...
+from presence_analyzer import main, utils
 
+MY_PATH = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, MY_PATH + '/../')
 
-TEST_DATA_CSV = os.path.join(
-    os.path.dirname(__file__), '..', '..', 'runtime', 'data', 'test_data.csv'
-)
+TEST_DATA_CSV = os.path.join(os.path.dirname(
+    __file__), '..', '..', 'runtime', 'data', 'test_data.csv')
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Style

File: src/presence_analyzer/tests.py [hunk 9]
```diff
@@ -42,6 +44,97 @@ def test_mainpage(self):
...
+        )
+        self.assertEqual(resp.status_code, 404)
+
+    def test_presence_mean_time_header(self):
+        """
+        Test presence mean time header.
+        """
+        resp = self.client.get('/static/mean_time_weekday.html')
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn(b'Presence mean time by weekday', resp.data)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: why b?

File: src/presence_analyzer/tests.py [hunk 10]
```diff
@@ -42,6 +44,97 @@ def test_mainpage(self):
...
+        """
+        resp = self.client.get('/static/presence_weekday.html')
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn(b'Presence by weekday', resp.data)
+
+    def weekday_parameter_correct(self):
+        """
+        Test presence by weekday api route parameter is correct.
+        """
+        weekdays = {
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: why you declare this here if you use it on the end of the function

File: src/presence_analyzer/tests.py [hunk 11]
```diff
@@ -42,6 +50,95 @@ def test_mainpage(self):
...
+            'Mon': 0,
+            'Tue': 30047,
+            'Wed': 24465,
+            'Thu': 23705,
+            'Fri': 0,
+            'Sat': 0,
+            'Sun': 0,
+        }
+        data = json.loads(resp.data)
+        self.assertDictEqual(weekdays, dict(data))
```
Comment (piotrgrundas) [inline src/presence_analyzer/tests.py]: Won't it return dict already?:)

File: src/presence_analyzer/tests.py [hunk 12]
```diff
@@ -42,6 +50,95 @@ def test_mainpage(self):
...
+            'Mon': 0,
+            'Tue': 30047.0,
+            'Wed': 24465.0,
+            'Thu': 23705.0,
+            'Fri': 0,
+            'Sat': 0,
+            'Sun': 0,
+        }
+        data = json.loads(resp.data)
+        self.assertDictEqual(weekdays, dict(data))
```
Comment (piotrgrundas) [inline src/presence_analyzer/tests.py]: ditto

**Pull request merged**